source/includes/connection-snippets/x509-snippets/ruby-connection.rb

```diff
@@ -0,0 +1,14 @@
+# begin x509 connection
+require 'mongo'
+
+begin
+  client = Mongo::Client.new(
+    ['localmongo1:27017'],
+    auth_mech. mongodb_x509,
+    ssl: true,
+    ssl_cert: '/etc/certs/mongodb/client.pem',
+    ssl_key: '/etc/certs/mongodb/client.pem',
+    ssl_ca_cert: '/etc/certs/mongodb/ca.pem'
+  )
```

Comment:
We documented how to enable x509 for local testing in https://github.com/mongodb/mongo-ruby-driver/blob/master/spec/README.md#x509-authentication. Unless creation of the client using Ruby options is specifically desired, I think using the URI format is a better idea.

Reply:
From a developer perspective, I think most users would rather use an options structure. Their IDEs will give code hints to help prevent typos and offer suggestions, and API line documentation if it exists. 

Why is the URI format a better idea and different from the other drivers?

Reply:
@p-mongo to clarify, does Ruby's client constructor solicit URI options outside of the connection string using the same keys as would be used in the connection string? This came up in the PHP example, where @mongomoe was passing the CA file and cert via a constructor parameter with different names for these options in an associative array (since deprecated), and I suggested using another constructor parameter that used the unified URI option names (also an associative array). There, I opted _not_ to stuff everything into a single connection string for the sake of readability. See: https://github.com/mongodb/docs-ecosystem/pull/573/files#r350349009

I don't know if Ruby allows specifying the unified URI options via the same option names anywhere beyond the connection string (i.e. first parameter), but I want to make sure you and @terakilobyte are on the same page there.

Reply:
No, in Ruby URI options only apply when they are given in the URI. When given as Ruby hash keys the option names can be entirely different.

I don't know if I personally care which syntax is used - we support both - I am only under the impression that in general the URI syntax is favored these days.

Reply:
Also, almost no developers initialize Ruby client programmatically. Almost all applications use Mongoid. Therefore while the given code example illustrates which Ruby options can be set and what they are called, it's not something that 95% of users will directly copy/paste into their configuration (they would have to convert this to Mongoid yaml syntax). A URI is more easily copy pastable.